reduce k value to result count

tomhamer opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
we reduce k to equal the result count, which vastly improves search latency. Since we have moved to lucene we no longer see the error with inner hits not being returned

* **What is the current behavior?** (You can also link to an open issue here)
Search latency is very high due to a high k - this is a problem since we moved from NMSLIB to lucene based search.

* **What is the new behavior (if this is a feature change)?**
Reduces search lateny on large indexes by approx 20x. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no


* **Have unit tests been run against this PR?** (Has there also been any additional testing?)


* **Related Python client changes** (link commit/PR here)


* **Related documentation changes** (link commit/PR here)


* **Other information**:


* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

tomhamer committed:
reduce k value to result count

pandu-k reviewed (changes requested):
This change results in a failing unit test: https://github.com/marqo-ai/marqo/actions/runs/3671015792/jobs/6205945452#step:8:3873

This unit test is regarding protections for when result_count gets too big (over 10k). 

This hits the k limit of Marqo-os.

Suggestion:
    - Correct the test (perhaps by mocking Marqo-os via mocking a GET request)
    - Set the default for this limit to the Marqo-os limit (for the best user experience): https://github.com/marqo-ai/marqo/blob/bb54a78cba69e3102883f99124a592072199b3a4/src/marqo/tensor_search/configs.py#L35

Also, what are the implications for a large result_count (result_count > 500)? For small Ks this change would result in a speed-up, but latency is probably degraded for these large cases

pandu-k commented:
Code to test the impact of missing inner hits:

- An OpenSearch community member's [gist](https://gist.github.com/BM25-enthusiast/56b00d98926942db56f8ad31830e81c8)
- OpenSearch contributor's [gist](https://gist.github.com/jmazanec15/5718153948b4441367093c74863504ba)

pandu-k committed:
minimum result_count now set to 0.

Default max docs set to 10k in line with marqo-os

This is because Marqo-os requires k>= 0.

pandu-k commented:
Tested for missing inner hits. None retrieved

pandu-k commented:
Ran test suite, all passed
